File: 0728-Self Dividing Numbers/0728-Self Dividing Numbers.cpp
class Solution {
public:
    vector<int> selfDividingNumbers(int left, int right) {
        vector<int> result;
        for (int i = left; i <= right; ++i) {
            int num = i;
            bool valid = true;
            while (num) {
                int temp = num % 10;
                if (temp == 0 || i % temp) {
                    valid = false;
                    break;
                }
                num /= 10;
            }
            if (valid) {
                result.push_back(i);
            }
        }
        
        return result;
    }
};

File: 0701-0800/0728-Self Dividing Numbers/0728-Self Dividing Numbers.py
class Solution:
    def selfDividingNumbers(self, left: int, right: int) -> List[int]:
        def isSelfDividing(num):
            x = num
            while num:
                num, d = divmod(num, 10)
                if d == 0 or x % d:
                    return False
            return True
        return [num for num in range(left, right + 1) if isSelfDividing(num)]
